### DISCUSSION TITLE
Goto definition of dependencies in unmanagedJars

### DISCUSSION THREAD

**Discussion by richardpen** [scalameta/metals#2429] [Q&A]
Hi,

I'm new to scala and try to setup my dev environment with neovim. I have a project that uses lots of other dependencies which are built jar files. I added them into my project's `build.sbt` file via `unmanagedJars in Compile := jars("/path/to/jar")`. The code navigation works great within the project code itself, but when the code references a method defined in the dependencies(defined in the jar file), the code navigation returns error `no definition found`. Is this expected, or I missed something in my setup? Any information would be help, thanks!

**Comment by ckipp01** [comment]
Thanks for the question! I've gone ahead and transferred this over to the main Metals repo as this will work the same no matter the client. As far as I know, this should work as long as the jars are correctly captured in the export that bloop does during the `sbt bloopInstall`.  While attempting to reproduce this however, I'm unfamiliar with the way you use `jars` here? Is that exactly how you have it in your `build.sbt`? The only time I've every used `unmanagedJars` were by just dropping them in the `lib` dir or doing something like this format [taken from the docs](https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html#Unmanaged+dependencies):

```
Compile / unmanagedJars := Seq.empty[sbt.Attributed[java.io.File]]
```

**Reply by richardpen** [reply]
@ckipp01 Thanks for moving it here, in my `build.sbt` file, I have `unmanagedJars in Compile := file("/path/to/jar")`.
`sbt compile` and `sbt bloopInstall` all works fine(no error), and I even put the following in my `build.sbt` per [instruction](https://scalacenter.github.io/bloop/docs/build-tools/sbt#download-dependencies-sources)
```
bloopExportJarClassifiers in Global := Some(Set("sources"))
bloopAggregateSourceDependencies in Global := true
```
I did notice that the generated bloop configuration file doesn't contain module(within resolution) for dependencies specified in the `unmanagedJars`. So that makes me think whether it's something in bloop, as I'm pretty new to scala and the tools, hope someone who has experienced with these can give me some hints.

**Reply by tgodzik** [reply]
I don't think sources are possible for unmanaged jars currently, there is no way to determine what should be the source jars for jars added manually to the build definition. Bloop (and Metals) gets that data from sbt and since it's not avialable there, we would need to work around it or have it fixed in sbt. 

The alternative would be to implement the decompiler and show the decompiled jar to the user in case we are missing the source jars or attach sources manually like in https://github.com/scalameta/metals-feature-requests/issues/12
